### ISSUE TITLE
Question about evaluation variance and rollout count

### ISSUE THREAD

**Issue by SII-ZhangYiFei** [cvlab-columbia/videopolicy#5]
Hi, thanks again for releasing the code and checkpoints — really appreciate it.
I evaluated the released checkpoint using the official evaluation code with 50 episode rollouts per task, and I obtain an average success rate of ~58%, which is somewhat lower than the ~66% reported in the paper.
<details>
<summary>Per-task success rates (paper vs my reproduction)</summary>

| Category | Task | Ours (50) | Ours (300) | **My Experiment ** |
|--------|------|-----------|------------|---------------------|
| Pick & Place | PnPCabToCounter | 0.36 | 0.48 | **0.22 (11/50)** |
|  | PnPCounterToCab | 0.42 | 0.52 | **0.34 (17/50)** |
|  | PnPCounterToMicrowave | 0.52 | 0.22 | **0.32 (16/50)** |
|  | PnPCounterToSink | 0.44 | 0.48 | **0.40 (20/50)** |
|  | PnPCounterToStove | 0.58 | 0.54 | **0.62 (31/50)** |
|  | PnPMicrowaveToCounter | 0.44 | 0.28 | **0.24 (12/50)** |
|  | PnPSinkToCounter | 0.64 | 0.56 | **0.66 (33/50)** |
|  | PnPStoveToCounter | 0.64 | 0.70 | **0.38 (19/50)** |
| Doors | OpenSingleDoor | 0.68 | 0.70 | **0.76 (38/50)** |
|  | OpenDoubleDoor | 0.96 | 0.94 | **0.92 (46/50)** |
|  | CloseDoubleDoor | 0.98 | 0.90 | **0.92 (46/50)** |
|  | CloseSingleDoor | 1.00 | 0.90 | **1.00 (50/50)** |
| Drawers | OpenDrawer | 0.46 | 0.54 | **0.42 (21/50)** |
|  | CloseDrawer | 0.96 | 1.00 | **0.94 (47/50)** |
| Twisting Knobs | TurnOnStove | 0.30 | 0.50 | **0.26 (13/50)** |
|  | TurnOffStove | 0.06 | 0.04 | **0.08 (4/50)** |
| Turning Levers | TurnOnSinkFaucet | 0.84 | 0.76 | **0.80 (40/50)** |
|  | TurnOffSinkFaucet | 0.78 | 0.92 | **0.78 (39/50)** |
|  | TurnSinkSpout | 0.40 | 0.58 | **0.30 (15/50)** |
| Pressing Buttons | CoffeePressButton | 0.92 | 0.96 | **0.92 (46/50)** |
|  | TurnOnMicrowave | 0.92 | 0.96 | **0.94 (47/50)** |
|  | TurnOffMicrowave | 0.90 | 1.00 | **0.86 (43/50)** |
| Insertion | CoffeeServeMug | 0.76 | 0.70 | **0.58 (29/50)** |
|  | CoffeeSetupMug | 0.22 | 0.58 | **0.30 (15/50)** |
| — | **Avg. Task Success Rate** | **0.63** | **0.66** | **0.58** |

</details>

### Questions
I wanted to ask a few clarification questions:
1. From your experience, is an average success rate around ~58% common for a single 50-rollout evaluation due to variance?
2. Are there any extra evaluation steps or details that are important but easy to miss?
3. Is it trained with **50 demonstrations** or **300 demonstrations** (as reported in the paper)?

**Comment by artpli** [comment]
Same question. Hi @junbangliang, could you please share any thoughts here? Many thanks!

Best,
Peng

**Comment by junbangliang** [comment]
Thanks for checking out the paper, the checkpoints released are trained for 50 demonstrations and not 300 demonstrations. And in my evaluations sometimes it does have variation where I saw success rates fluctuate between 60% and 63%. I will rerun the original script when I submitted the paper to check if there are any missed details and will get back to you. 

Also, can you please check if any experiment output videos look like corrupted, I have seen other cases when replicating the same experiments but the Robocasa setup sometimes gives corrupted environments during eval, this seems to be dependent on the computer and is a known [issue](https://github.com/robocasa/robocasa/issues/114) in Robocasa.

**Comment by SII-ZhangYiFei** [comment]
Hi, thanks a lot for the clarification and for double-checking the original evaluation — really appreciate the help and the transparency.

I also wanted to ask a follow-up question regarding training time / number of steps. I started training using the released code on a single machine with 8× H100 GPUs, and I’m seeing logs like:
Epoch 0: 7% | 759/10862 [18:38<4:08:10, 0.68 it/s]

At this speed, reaching ~368k training steps (as reported in the paper) would take a bit over 6 days of wall-clock time on my setup just on stage 1. I just wanted to ask for clarification on this point:

Does the reported 368,866 steps correspond to a similar single-node setup, or was training done with more nodes / a different parallelization setup?

Are there any speed-related configurations (e.g., dataloader settings, precision, gradient accumulation, etc.) that are important for matching the reported training time?

Just want to make sure I’m interpreting the training setup correctly and not missing something obvious. Thanks again for releasing the code and checkpoints — it’s been really helpful for understanding the method in detail.

**Comment by junbangliang** [comment]
For the training time, the training setup I released is identical to the training I used. I trained on a single node of 8xA100 back then and it took about 14 days, and with a 8xH100 node it should be the expected time it takes for the training.

The setup I released is training with mixed-16 PyTorch Lightning configuration and it is the precision I used for training. I observed very mirror difference with fp32 training and it could be due to the random variation with the Robocasa evaluation.

There are several points I would recommend to improve the training details if you are interested:
1. Maybe try using a Muon optimizer like in [Hunyuan 1.5](https://github.com/Tencent-Hunyuan/HunyuanVideo-1.5), where it should improve the model convergence.
2. Try to train with a bigger batch size, I think training with a batch size of 32 is too small in hindsight, increasing the batch size should improve model convergence. Perhaps try training with a varying batch size schedule rather than a fixed schedule.
3. I have noticed that increasing video resolution from 128x128 to 256x256 when training the SVD checkpoint from scratch would significantly improve model performance, but we are limited by compute to try higher resolutions.
4. A better architecture for the video adaptor (self.vision_encoder) and the action decoder (self.pose_pred_net) with a transformer-based model rather than CNN based model should improve the model performance in hindsight.
5. Training with the 300 demo with 368k step resulted in the same performance compared to training with 50 demos. Double the steps are needed for the 300 demo dataset to show improvement in performance. This could be due to training with the same batch size, where the gradient is too noisy with a small batch size to show significant difference. 
6. There is a bug with logging if gradient accumulation is set above 1, if you encounter that bug please set image_logger to be disabled in the model training config.

**Comment by SII-ZhangYiFei** [comment]
Thank you again for the incredibly detailed and insightful response. Your retrospective notes on batch size, optimizer choice, and architectural limitations are especially valuable.

I’m writing to share a concrete observation from my own training run and to ask for your intuition, given your hands-on experience with the model.

I trained a GR1 policy using the released pipeline, starting from the SVD checkpoint, under the following setup:

Dataset: nvidia/PhysicalAI-Robotics-GR00T-X-Embodiment-Sim

Demos: 50 per task

Steps: Stage 1: ~80k, Stage 2: ~20k

View setup: Single-view prediction (predicting 1+8 frames instead of 25)

Evaluation: robocasa-gr1-tabletop-tasks

The resulting policy shows two clear issues during rollouts:

Very low success rate (~1.67% average).

Strong end-effector shaking/jitter, with noticeably unsmooth and unstable action trajectories, even in simple task setups.

Given your deep familiarity with the model’s behavior, I’d be grateful for your thoughts on a few points:

Did you encounter similar low success rates or pronounced action jitter during your own training or evaluation phases?

Does this behavior align with what you’d expect from (1) single-view prediction, (2) fewer steps, or (3) robot-specific control and action-scaling mismatches?

I’m trying to discern whether this is an expected outcome given the constraints I used, or if it might point to a deeper configuration issue. Your guidance would be immensely helpful.

Thanks again for your openness and for sharing this codebase—it’s a fantastic resource for digging into these questions.

**Comment by junbangliang** [comment]
Hi,

Thanks for getting back. I have the following observations that might be able to help you: 
1. When I trained the multitask policy, I remembered that not until 170k steps into the training I started seeing successful rollouts where the video generation actually start to match the actions outputs from the action decoder. I would suggest training with a single task to quickly check if the single task policy can converge more quickly. I vaguely remembered that single task training only takes around 80k steps to converge. 
2. Also, I found that you need to train the action decoder the same number of steps as the video model in stage 1 training, otherwise the action decoder won't converge and harm model performance. I haven't tried training with a single view, but I tried training with 2 views and it works just fine. 
3. In Robocasa, there is issues with the actions if not setup properly, I ran into issues when I tried to change the default simulation environment, so I ended up not changing the default environment and leave it as it is. You can quickly train a diffusion policy to see if you have the simulation environment setup correctly. I think it could be that you are not normalizing the action correctly.

**Comment by SII-ZhangYiFei** [comment]
Thank you so much for the thoughtful and detailed response — this is exactly the guidance I needed. The points on convergence timing, action decoder training, and environment setup are incredibly helpful. I really appreciate you taking the time to share these insights.

**Comment by junbangliang** [comment]
Hi,

I went back to the checkpoints to run many experiments, and found significant differences across runs even for the same checkpoint, this is likely due to randomness of the Robocasa environments. I also observed noticeable performance differences between consecutive checkpoints, this is likely because I trained with a constant learning rate and did not use a learning-rate cooldown.

To address this, I replaced stage_2_trained_step_369k.ckpt with another checkpoint near 369k steps and hopefully you see better and less variations in performance. I evaluated this replacement checkpoint using five consecutive experiments to measure its variation, and it appears to be slightly more stable than the previous one.

You can re-download the updated checkpoint set with:
`wget https://videopolicy.cs.columbia.edu/assets/checkpoints.zip`

I also released the five example experiments, available at:
`wget https://videopolicy.cs.columbia.edu/assets/example_experiments.zip`
